src/droplet.F

```diff
@@ -3109,16 +3116,24 @@ subroutine droplet_injection(pdata,xf,yf,dt,time_since_inject,randomNumbers,mtim
 
       if (tot_after_reintro .lt. nparcelsMax) then   !Only add particles if the total number injected
                                                      !does not make the global exceed nparcelsMax
-         do np=nparcelsLocalActive+1,nparcelsLocalActive+my_reintro
+         !$acc data copyin(dFssum,binsdata)
+         !$acc parallel loop gang vector default(present) reduction(+:mass_injected)
+         do j=1,my_reintro
+            np=nparcelsLocalActive+j
             if (random_droplets) then
+               ! initialize a new random number generator for each
+               ! injected droplet
+               local_seed = (1000 * int(mtime)) + (10000*myid) + j
+               randomNumbers = new_RandomNumberSequence(seed = local_seed)
```

Comment:
Not sure if we want to shuffle the seeds every time we call the `droplet_injection` subroutine. Does it make a difference here?

Reply:
My motivation is to make sure we have a deterministic sequence of random numbers.  If we do it only once before the loop, then there will potentially be duplicate droplets on each set of threads, right?

Reply:
Thanks @johnmauff . After a second thought, I think you are right that without setting a unique seed for each CUDA thread, the result may not be deterministic because the sequence of pulling random number by each CUDA thread is not guaranteed.